pkg/download/protocol.go

```diff
@@ -74,19 +77,27 @@ func (p *protocol) List(ctx context.Context, mod string) ([]string, error) {
 	ctx, span := observ.StartSpan(ctx, op.String())
 	defer span.End()
 
+	// If the download mode is None, we won't be downloading anything from
+	// the internet anyway, so we should only examine local storage.
+	localOnly := p.df.Match(mod) == mode.None
```

Comment:
@cobalt77 I think we need a different signal to determine when to not look to the internet. `mode.None` means that a 404 should be returned for this module, unless I'm misunderstanding something?

Reply:
Thought a lot about this. Let's say I have the following setup:
- Offline environment, and a self-hosted git server with some go modules
- Athens with file based storage and a dynamic config where all modules except my.git.server.com/* get download mode "None"

If I go get github.com/something, that will use download mode None and treat the contents of the athens cache as ground truth, and get me the latest of the available modules.
If I go get my.git.server.com/somethingelse, athens will go out to my git server for the module, and not assume that what's in cache is the latest.

This will work as expected with this change. But it changes what None does, and maybe there is a scenario where a 404 would be preferable. Or in online environments, athens treating the cache as ground truth _sometimes_ but not all the time could lead to nondeterministic behavior or other weird issues. 

What do you think about keeping None as-is (always 404 if it doesn't exist), and adding a new separate download mode that will do this?